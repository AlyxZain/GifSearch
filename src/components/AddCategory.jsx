import { useState } from "react";

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState("");

  const oninputChange = ({ target }) => {
    setInputValue(target.value);
  };

  const onSubmit = (event) => {
    const newCategory = inputValue.trim();

    event.preventDefault();
    if (newCategory.length <= 0) return;
    setInputValue("");

    onNewCategory(newCategory);
  };

  return (
    <form onSubmit={onSubmit}>
      <input
        type='text'
        placeholder='Buscar Gifs'
        value={inputValue}
        onChange={oninputChange}
      />
    </form>
  );
};
