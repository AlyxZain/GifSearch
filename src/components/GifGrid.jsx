import { GifItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";

export const GifGrid = ({ category, del }) => {
  const { images, isLoading } = useFetchGifs(category);

  //borrar images

  return (
    <>
      <h3>{category}</h3>

      <div className='card-grid'>
        {images.map((images) => (
          <GifItem key={images.id} {...images} />
        ))}
      </div>
    </>
  );
};
