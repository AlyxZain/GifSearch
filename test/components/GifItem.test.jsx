import { render } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("Prueba en <GifItem />", () => {
  const title = "Goku";
  const url = "https://media.giphy.com/media/CY3A9zOlZR8uhFbeok/giphy.gif";

  test("Debe hacer match con el snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("Debe de mostrar la imagen con el Url y el alt indicado", () => {
    render(<GifItem title={title} url={url} />);

    const { src, alt } = document.querySelector("img");
    expect(src).toBe(url);
    expect(alt).toBe(alt);
  });

  test("Debe de mostrar el titulo indicado", () => {
    render(<GifItem title={title} url={url} />);
    expect(document.querySelector("p").textContent).toBe(title);
  });
});
