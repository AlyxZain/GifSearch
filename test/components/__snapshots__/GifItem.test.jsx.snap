// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <GifItem /> Debe de mostrar el componente correctamente 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Goku"
      src="https://media.giphy.com/media/CY3A9zOlZR8uhFbeok/giphy.gif"
    />
    <p>
      Titulo del GIF
    </p>
  </div>
</div>
`;

exports[`Prueba en <GifItem /> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Goku"
      src="https://media.giphy.com/media/CY3A9zOlZR8uhFbeok/giphy.gif"
    />
    <p>
      Goku
    </p>
  </div>
</div>
`;
